src/io/inputs/model-converter/convertorVisitor.cpp

```diff
@@ -104,12 +104,25 @@ Expressions::NodeRegistry convertExpressionToNode(const std::string& exprStr,
     {
         return {};
     }
+
+    // ANTLR setup
     antlr4::ANTLRInputStream input(exprStr);
     ExprLexer lexer(&input);
     antlr4::CommonTokenStream tokens(&lexer);
     ExprParser parser(&tokens);
 
-    ExprParser::ExprContext* tree = parser.expr();
+    // error handling
+    auto errorListener = std::make_shared<ErrorListener>();
```

Comment:
You actually don't need a `std::shared_ptr` here, since the object will be deallocated at the end of scope


```suggestion
    ErrorListener errorListener;
```